docs: go tool note (#5549)
golangci/golangci-lint: docs/src/docs/welcome/install.mdx

go install github.com/golangci/golangci-lint/v2/cmd/golangci-lint@{.LatestVersion}
```

<details>
<summary>`go tool` usage recommendations</summary>

We don't recommend using `go tool`.

> But if you want to use `go tool` to install and run golangci-lint (once again we don't recommend that),

Reviewer: This mentioned in the previous sentence, no need to repeat: 
```suggestion
But if you want to use `go tool` to install and run golangci-lint,
```

Author: The repetition is intentional.

the best approach is to use a dedicated module or module file to isolate golangci-lint from other tools or dependencies.

> This approach avoid to modifying your project dependencies and the golangci-lint dependencies.

Reviewer: ```suggestion
This approach avoids modifying your project dependencies and the golangci-lint dependencies.
```

**⚠️ IMPORTANT ⚠️: You should never update golangci-lint dependencies manually.**

> **Method 1 (dedicated module file)**

Reviewer: We can avoid parentheses:
```suggestion
**Method 1: dedicated module file**
```

or
```suggestion
**Method 1. Dedicated module file**
```
